### ISSUE TITLE
banners ad as widget issue

### ISSUE THREAD

**Issue by leatdev** [AppLovin/AppLovin-MAX-Flutter#32]
### MAX Plugin Version

2.0.0

### Flutter Version

3.0.5

### Device/Platform Info

Android 11

### Current Behavior

rendering banners ad as widget not working, please fix the issue

### How to Reproduce

add banners ad as widget, ad will not show

**Comment by thomasmso** [comment]
@leatdev - Do you mind enabling and providing [verbose logs](https://dash.applovin.com/documentation/mediation/flutter/getting-started/advanced-settings#enable-verbose-logging)?

**Comment by leatdev** [comment]
@thomasmso
<img width="894" alt="error" src="https://user-images.githubusercontent.com/77149638/180297165-7371fc42-0a7a-463c-b3ce-01a0d967ba4d.png">

**Comment by cmlee-danbi** [comment]
Same here

Flutter 3.0.5
Dart 2.17.6

**Comment by ahammamlho** [comment]
same problem 

Flutter 3.3.0 • channel stable 
Tools • Dart 2.18.0 • DevTools 2.15.0

**Comment by kkorgol** [comment]
Same here. Please resolve this issue

**Comment by thomasmso** [comment]
Hey yall - I'd love to help, but I need someone to enable and send over our plugin's [verbose logs](https://dash.applovin.com/documentation/mediation/flutter/getting-started/advanced-settings#enable-verbose-logging) for us to investigate.

**Comment by kkorgol** [comment]
My verbose logs: [verbose_logs_2022-09-29.txt](https://github.com/AppLovin/AppLovin-MAX-Flutter/files/9679229/verbose_logs_2022-09-29.txt)

**Comment by thomasmso** [comment]
I see this:

```
E/AppLovinSdk(20678): [AppLovinMAX] Failed to create MaxAdView widget - please ensure the AppLovin MAX plugin has been initialized by calling 'AppLovinMAX.initialize(...);'!
```

which means the widget should be attached later in your session - not immediately upon app launch

**Comment by kkorgol** [comment]
I'm sure that AppLovin MAX is initialized. This is how widget is loaded:

```
FutureBuilder<bool>(
            future: AppLovinMAX.isInitialized(),
            builder: (BuildContext context, AsyncSnapshot<bool> snapshot) {

              print("isInitialized: " + snapshot.data.toString() );

              return snapshot.data != null && snapshot.data
                  ? MaxAdView(
                  adUnitId: "XXXXXXX",
                  adFormat: AdFormat.banner,
                  listener: AdViewAdListener(onAdLoadedCallback: (ad) {
                    logStatus('Banner widget ad loaded from ' + ad.networkName);
                  }, onAdLoadFailedCallback: (adUnitId, error) {
                    logStatus('Banner widget ad failed to load with error code ' + error.code.toString() + ' and message: ' + error.message);
                  }, onAdClickedCallback: (ad) {
                    logStatus('Banner widget ad clicked');
                  }, onAdExpandedCallback: (ad) {
                    logStatus('Banner widget ad expanded');
                  }, onAdCollapsedCallback: (ad) {
                    logStatus('Banner widget ad collapsed');
                  }))
                  : Container();
            }
        )
```

When AppLovin is not loaded then Container is showing. If AppLovin is laoded then MaxAdView. Even print with 'isInitialized' returns true (this print has been commented for verbose_logs_2022-09-29.txt purpose). Please check [verbose_logs_2022-09-30.txt](https://github.com/AppLovin/AppLovin-MAX-Flutter/files/9680904/verbose_logs_2022-09-30.txt) with uncommented this line. isInitialized is true and then MaxAdView returns error. This look very strange

**Comment by thomasmso** [comment]
Are you able to reproduce this in our example app? If I have a reproducible case locally I can solve it.

**Comment by ts8ta** [comment]
We had pretty much the same issue, described here: https://github.com/AppLovin/AppLovin-MAX-Flutter/issues/31

Although we were not able to reproduce in the example app, we were not able to integrate it in our application and we were also getting the info that the plugin was not initialized.

**Comment by kkorgol** [comment]
I finally reproduced this issue on example app.

In my case error is throw after call FirebaseMessaging.onBackgroundMessage(_firebaseMessagingBackgroundHandler) method from firebase_messaging package

firebase_messaging package is a crucial package for everyone that want to use Firebase Cloud Messaging so in my opinion this issue is very important because we can not show notifications and have ads in our app.

Steps to reproduce:
1. Clone https://github.com/kkorgol/AppLovin-MAX-Flutter.git
2. Checkout [banners-ad-as-widget-issue](https://github.com/kkorgol/AppLovin-MAX-Flutter/tree/banners-ad-as-widget-issue) branch
3. Run application
4. Click on Show Widget Banner or Show Widget MREC button

Result:
Error is thrown in Console

Commit only: https://github.com/kkorgol/AppLovin-MAX-Flutter/commit/11f50e707b86a43a1f984c2e69a3429c39162f16

**Comment by thomasmso** [comment]
@kkorgol - Thanks for the reproduction case. Do you have an account manager at AppLovin? Our engineering pipeline is quite full at the moment and we can only prioritize this investigation under specific circumstances.

**Comment by kkorgol** [comment]
If you mean account in dash.applovin.com then yes, I have. If you mean different type of account then I do not have.

I highly recommended to check by your devs [this](https://github.com/AppLovin/AppLovin-MAX-Flutter/pull/35) pull request. I merged this to my fork and looks like banner works well in my Android production app

**Comment by thomasmso** [comment]
Thanks for checking it out, I'll go ahead and merge that PR in, and push a new release.

**Comment by thomasmso** [comment]
Thanks for checking it out, I'll go ahead and merge that PR in, and push a new release.

**Comment by thomasmso** [comment]
Thanks for checking it out, I'll go ahead and merge that PR in, and push a new release.

**Comment by thomasmso** [comment]
Released in v2.3.1.
